### PR TITLE
CoW: Fix deprecation warning for chained assignment

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I hate the SettingWithCopyWarning mechanism...

2 cases were fixed here:

- the parent that holds the item_cache might be dead, e.g. the ref count is not increased. We have to check this when looking at the cacher
- iloc does not populate the item_cache but it sets _cacher on the new object, so we have to check if we are actually part of the item_cache

xref https://github.com/pandas-dev/pandas/issues/56019

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
> I hate the SettingWithCopyWarning mechanism...

Oh I am looking forward to ripping out all that code ;)

**Comment by phofl** [comment]
Yes me too, I call dips on doing that :)

File: pandas/tests/copy_view/test_chained_assignment_deprecation.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        ("fillna", (1,)),
+        ("interpolate", ()),
+        ("bfill", ()),
+        ("ffill", ()),
+    ],
+)
+def test_methods_iloc_getitem_item_cache(func, args, using_copy_on_write):
+    df = DataFrame({"a": [1, 2, 3], "b": 1})
+    ser = df.iloc[:, 0]
+    getattr(ser, func)(*args, inplace=True)
```
Comment (jorisvandenbossche) [inline pandas/tests/copy_view/test_chained_assignment_deprecation.py:39]: This should raise a warning?

Comment (jorisvandenbossche) [reply pandas/tests/copy_view/test_chained_assignment_deprecation.py:39]: (or is that something we haven't implemented yet)

Comment (phofl) [reply pandas/tests/copy_view/test_chained_assignment_deprecation.py:39]: That should raise but we haven’t done that yet I think, but it shouldn’t raise a chained assignment warning, which is what happens on main right now 

the implementation will look similar to #56168

**Review by jorisvandenbossche — APPROVED** [review]
Thanks a lot! 
Merging this so I can use it in my chained assignment PR

**Pull request merged**